gswrap/__init__.py

```diff
@@ -693,7 +687,7 @@ def _upload(self,
 
         # None is ThreadPoolExecutor max_workers default. 1 is single-threaded
         max_workers = None if multithreaded else 1
-        futures = []  # type: List[concurrent.futures.Future]
+        futures = []  # List[concurrent.futures.Future]
```

Comment:
Actually, you can refactor this code and get away with setting the type yourself. I think it would also improve readability a lot. Here's a sketch:

```
blobs_paths = []  # type: List[Tuple[Blob, pathlib.Path]]
for file_name in upload_files:
    ...
    blobs_paths.append((blob, pth))

with Executor as executor:
    futures = [executor.submit(..., blob=blob, path=pth, ...)
                    for blob, pth in blobs_paths]
    for future in futures:
        _ = future.result()
```

This way you clearly separate the code that determines the parameters from execution, so the reader can focus on a single aspect at the time.

Reply:
```
def generate_upload_files()->Iterable[Tuple[Blob, pathlib.Path]]:
    for file_name in upload_files:
        ...
        yield blob, pth

with Executor as executor:
    futures = [executor.submit(...), for blob, pth in generate_blobs_paths()]
    for future in futures:
        _ = future.result()
```